### ISSUE TITLE
Consider implementing deploy-maven using rules_jvm_external

### ISSUE THREAD

**Issue by lolski** [graknlabs/bazel-distribution#237]
Based on the [docs](https://github.com/bazelbuild/rules_jvm_external#publishing-to-external-repositories), `rules_jvm_external` seem to have the ability to deploy Maven snapshots. We should consider using it since we can trim down our implementation significantly.

Here's the excerpt from the docs linked above:

---

**Publishing to External Repositories**
In order to publish an artifact from your repo to a maven repository, you must first create a java_export target. This is similar to a regular java_library, but allows two additional parameters: the maven coordinates and an optional template file to use for the pom.xml file.

```
# user_project/BUILD
load("@rules_jvm_external//:defs.bzl", "java_export")

java_export(
  name = "exported_lib",
  maven_coordinates = "com.example:project:0.0.1",
  pom_template = "pom.tmpl",  # You can omit this
  srcs = glob(["*.java"]),
  deps = [
    "//user_project/utils",
    "@maven//:com_google_guava_guava",
  ],
)
```
In order to publish the artifact, use bazel run:

```
bazel run --define "maven_repo=file://$HOME/.m2/repository" //user_project:exported_lib.publish
```

Or, to publish to (eg) Sonatype's OSS repo:

```
bazel run --stamp \
  --define "maven_repo=https://oss.sonatype.org/service/local/staging/deploy/maven2" \
  --define "maven_user=example_user" \
  --define "maven_password=hunter2" \
  --define gpg_sign=true \
  //user_project:exported_lib.publish`
```

When using the gpg_sign option, the current default key will be used for signing, and the gpg binary needs to be installed on the machine.

**Comment by haikalpribadi** [comment]
Merged into issue #243

**Comment by lolski** [comment]
This issue has been closed since it is too specific and does not contain the relevant context.

Please refer to https://github.com/graknlabs/bazel-distribution/issues/243 which better describes the overall problem that we want to solve.
